### PR TITLE
Add architecture page

### PR DESCRIPTION
**Pull request by masayag**
Selection_145

### CODE REVIEW

File: content/docs/architecture/index.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: "Architecture"
+---
+The Orchestrator architecture comprises several integral components, each contributing to the seamless execution and management of workflows. Illustrated below is a breakdown of these components:
+
```
Comment (gciavarrini) [inline content/docs/architecture/index.md:5]: nit: I would suggest highlighting the keywords in the bullet list to enhance the readability.

File: content/docs/architecture/index.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+- Backstage: Serving as the primary interface, Backstage fulfills multiple roles:
+  - Orchestrator Plugins: Both frontend and backend plugins are instrumental in presenting deployed workflows for execution and monitoring.
```
Comment (gciavarrini) [inline content/docs/architecture/index.md:7]: WDYT to add a link to the relevant docs?
This comment applies to other items in this list.

File: content/docs/architecture/architecture-diagram.png
Comment (gciavarrini) [inline content/docs/architecture/architecture-diagram.png:1]: very nit (feel free to ignore): the green arrow covers PostgreSQL.

File: content/docs/architecture/index.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- Data Index Service: This serves as a repository for workflow definitions, instances, and their associated jobs. It exposes a GraphQL API, utilized by the Orchestrator backend plugin to retrieve workflow definitions and instances.
+- Job Service: Dedicated to orchestrating scheduled tasks for workflows.
+- OpenShift Serverless: This operator furnishes serverless capabilities essential for workflow communication. It employs Knative eventing to interface with the Data Index service and leverages Knative functions to introduce more intricate logic to workflows.
+- AMQ Streams (Strimzi/Kafka): While not presently integrated into the deployment's current iteration, this operator is crucial for ensuring the reliability of the eventing system.
+- KeyCloak: Responsible for authentication and security services within applications. While not part of the current deployment version, it is essential for enhancing security measures.
+
+![Architecture Diagram](./architecture-diagram.png)
```
Comment (gciavarrini) [inline content/docs/architecture/index.md:17]: can you increase the size a bit?

Comment (masayag) [reply content/docs/architecture/index.md:17]: couldn't find a direct way of doing so, need to research a bit and maybe enable some hook to support it:
https://gohugo.io/getting-started/configuration-markup/#image-markdown-example

Comment (gciavarrini) [reply content/docs/architecture/index.md:17]: Ok, we can address this later :)

**Comment by gciavarrini** [comment]
Overall lgtm, if you can move up the `Architecture` section in the left column would be a plus

**Review by gciavarrini — APPROVED** [review]

**Pull request merged**